bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -247,6 +250,23 @@ private String addRule0(Rule rule, String identity) {
     }
 
     /**
+     * Validates ids of modules. The module id must not contain dot.
+     *
+     * @param modules list of trigger, condition and action modules
+     * @throws IllegalArgumentException when a module id contains dot.
+     */
+    private void validateModules(List<Module> modules) {
+        for (Module m : modules) {
+            String mId = m.getId();
+            if (mId == null || mId.indexOf('.') > -1) {
```

Comment:
It is more reasonable to do mId.matches("[A-Za-z0-9_-]*") instead of just searching for dots in the String. With this approach you also exclude any other special character.